src/tests/system/tests/test_ipa.py

```diff
@@ -596,3 +597,925 @@ def test_ipa__idview_fails_to_apply_on_ipa_master(ipa: IPA):
+      3. Run `ipa hbactest` for user3 (admins group) — expect access denied.
+      4. Run `ipa hbactest` for user4 (no group) — expect access denied.
+      5. Validate SSH login access for all users.
+      6. Remove user1 from developers group.
+      7. Restart SSSD and validate user1 access is now denied.
+
+    :expectedresults:
+      1. user1 access granted and matched with developers_ssh_access rule.
+      2. user2 access granted and matched with developers_ssh_access rule.
+      3. user3 access denied as not in developers group.
+      4. user4 access denied as not in any group.
+      5. user1 and user2 login successful; user3 and user4 logins fail.
+      6. user1 removed from developers group successfully.
+      7. user1 access denied after group removal; user2 still has access.
+
+    :customerscenario: False
+    """
+    u1 = ipa.user("user1").add()
+    u2 = ipa.user("user2").add()
+    u3 = ipa.user("user3").add()
+    ipa.user("user4").add()
+
+    dev_group = ipa.group("developers").add()
+    admin_group = ipa.group("admins")
+
+    dev_group.add_members([u1, u2])
+    admin_group.add_members([u3])
+
+    ipa.hbac("allow_all").disable()
+
+    developers_ssh_rule = ipa.hbac("developers_ssh_access").create(
+        description="SSH access for developers group", groups="developers", hosts="client.test", services="sshd"
+    )
+
+    hbactest_user1 = developers_ssh_rule.test(user="user1", host="client.test", service="sshd")
+    assert hbactest_user1["access_granted"], "user1 should have access via developers group"
+    assert (
+        "developers_ssh_access" in hbactest_user1["matched_rules"]
+    ), "developers_ssh_access rule should match for user1"
+
+    hbactest_user2 = developers_ssh_rule.test(user="user2", host="client.test", service="sshd")
+    assert hbactest_user2["access_granted"], "user2 should have access via developers group"
+    assert (
+        "developers_ssh_access" in hbactest_user2["matched_rules"]
+    ), "developers_ssh_access rule should match for user2"
+
+    hbactest_user3 = developers_ssh_rule.test(user="user3", host="client.test", service="sshd")
+    assert not hbactest_user3["access_granted"], "user3 should be denied (not in developers group)"
+
+    hbactest_user4 = developers_ssh_rule.test(user="user4", host="client.test", service="sshd")
+    assert not hbactest_user4["access_granted"], "user4 should be denied (no group membership)"
+
+    client.sssd.restart()
+
+    assert client.auth.ssh.password("user1", "Secret123"), "user1 SSH should succeed"
+    assert client.auth.ssh.password("user2", "Secret123"), "user2 SSH should succeed"
+    assert not client.auth.ssh.password("user3", "Secret123"), "user3 SSH should fail"
+    assert not client.auth.ssh.password("user4", "Secret123"), "user4 SSH should fail"
+
+    dev_group.remove_member(u1)
+
+    client.sssd.restart()
+    time.sleep(10)  # Need sleep to make it passed
```

Comment:
Any idea why this fails without the sleep? Can you try with a shorter sleep  like 5 seconds?

Reply:
I tested with 5 seconds, but it's still failing with 5 seconds and passing with 10s.